### PR TITLE
modules: Add Sound Open Firmware module

### PR DESCRIPTION
**Pull request by nashif**
Add the Sound Open Firmware project as a Zephyr module under
modules/audio/sof.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

File: samples/audio/sof/prj.conf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CONFIG_SOF=y
+CONFIG_SMP=n
```
Comment (lgirdwood) [inline samples/audio/sof/prj.conf:2]: SOF has now aligned with Zephyr so should now be ```CONFIG_MULTICORE```

File: samples/audio/sof/sample.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  description: Audio with SOF
+  name: sof
+common:
+    tags: audio
+tests:
+  sample.audio.sof:
+    tags: sof
+    build_only: true
+    integration_platforms:
+      - intel_adsp_cavs15
```
Comment (lgirdwood) [inline samples/audio/sof/sample.yaml:11]: @lyakh fyi - can you add cavs18 - 25 here once you are happy all is working.

**Review by lgirdwood — CHANGES_REQUESTED** [review]
All looks good, just need to fix the multicore CONFIG.

**Review by lyakh — APPROVED** [review]

**Comment by nashif** [comment]
> Agreed, but dont we need to remove the unused `CONFIG_SMP` here ? SOF used to use this but now uses the Zephyr `CONFIG_MULTICORE` which should be `n` atm.

Zephyr does not have CONFIG_MULTICORE, we do have CONFIG_SMP

File: samples/audio/sof/sample.yaml (hunk 2 above)
Comment (lyakh) [reply samples/audio/sof/sample.yaml:11]: @lgirdwood is this for CI? And it seems to be only testing building, so, yes, should be good to add others too. @nashif in an additional PR or in this one?

**Comment by lgirdwood** [comment]
> > Agreed, but dont we need to remove the unused `CONFIG_SMP` here ? SOF used to use this but now uses the Zephyr `CONFIG_MULTICORE` which should be `n` atm.
> 
> Zephyr does not have CONFIG_MULTICORE, we do have CONFIG_SMP

ok - it looks like we made that change in error as SOF also used CONFIG_SMP in the past. This can be reverted for SOF v1.7. @lyakh fyi - need to realign with Zephyr here (I've no recollection why we made this change now).

**Review by lgirdwood — APPROVED** [review]

**Comment by lyakh** [comment]
> > > Agreed, but dont we need to remove the unused `CONFIG_SMP` here ? SOF used to use this but now uses the Zephyr `CONFIG_MULTICORE` which should be `n` atm.
> > 
> > 
> > Zephyr does not have CONFIG_MULTICORE, we do have CONFIG_SMP
> 
> ok - it looks like we made that change in error as SOF also used CONFIG_SMP in the past. This can be reverted for SOF v1.7. @lyakh fyi - need to realign with Zephyr here (I've no recollection why we made this change now).

@lgirdwood @nashif ok, but we cannot add an `SMP` config to SOF as long as we have to support both native and Zephyr building, right? Same for `MP_NUM_CPUS` which is equivalent to our `CORE_COUNT` or `MAX_CORE_COUNT`. We could rename `MULTICORE` to `SOF_SMP` and do something like
```
config SOF_SMP
    default SMP if ZEPHYR
```
if that's even possible. But in general until we completely switch over to Zephyr we need that duplication?

**Comment by lgirdwood** [comment]
> > > > Agreed, but dont we need to remove the unused `CONFIG_SMP` here ? SOF used to use this but now uses the Zephyr `CONFIG_MULTICORE` which should be `n` atm.
> > > 
> > > 
> > > Zephyr does not have CONFIG_MULTICORE, we do have CONFIG_SMP
> > 
> > 
> > ok - it looks like we made that change in error as SOF also used CONFIG_SMP in the past. This can be reverted for SOF v1.7. @lyakh fyi - need to realign with Zephyr here (I've no recollection why we made this change now).
> 
> @lgirdwood @nashif ok, but we cannot add an `SMP` config to SOF as long as we have to support both native and Zephyr building, right? Same for `MP_NUM_CPUS` which is equivalent to our `CORE_COUNT` or `MAX_CORE_COUNT`. We could rename `MULTICORE` to `SOF_SMP` and do something like
> 
> ```
> config SOF_SMP
>     default SMP if ZEPHYR
> ```
> 
> if that's even possible. But in general until we completely switch over to Zephyr we need that duplication?

The PR is good to go.  SOF needs to chnage back to CONFIG_SMP and this must be set to ```n``` until @andyross is happy with the SMP support on Intel xtensa DSP.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**